test/core/event_engine/windows/self_deleting_closure.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_TEST_CORE_EVENT_ENGINE_WINDOWS_SELF_DELETING_CLOSURE_H
+#define GRPC_TEST_CORE_EVENT_ENGINE_WINDOWS_SELF_DELETING_CLOSURE_H
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/functional/any_invocable.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class SelfDeletingClosure : public EventEngine::Closure {
```

Comment:
This is probably something we want in test/core/event_engine/util?

Reply:
Migrated to test/core/event_engine/common_closures.h and added tests (now with more custom destructors!)